code/world.dm

```diff
@@ -754,15 +378,27 @@ var/world_topic_spam_protect_time = world.timeofday
 
 #undef FAILED_DB_CONNECTION_CUTOFF
 
-/world/proc/do_topic_spam_protection(var/addr, var/key)
-	if (!config.comms_password || config.comms_password == "")
-		return "No comms password configured, aborting."
+/world/proc/do_auth_check(var/addr, var/key, var/function)
+	addr = sanitizeSQL(addr)
+	key = sanitizeSQL(key)
+	function = sanitizeSQL(function)
+	var/DBQuery/query = dbcon.NewQuery(

```

Comment:
The text here is formatted badly, and giving compile time errors. If you want multiline text, easiest way is to to do this:
```
var/DBQuery/query = dbcon.NewQuery({"Stuff
More stuff
Even more stuff"})
```